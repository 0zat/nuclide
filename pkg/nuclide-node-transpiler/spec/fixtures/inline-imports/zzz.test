import 'x';

//---

import Foo1A from 'x1';
import * as bar1A from 'x1';
import Foo1B, * as bar1B from 'x1';

Foo1A();
Foo1A.quux();

XYZ(Foo1A);
XYZ(Foo1A.quux);

new Foo1A();
new Foo1A.quux();

bar1A();
bar1A.quux();

XYZ(bar1A);
XYZ(bar1A.quux);

new bar1A();
new bar1A.quux();

Foo1B();
Foo1B.quux();

XYZ(Foo1B);
XYZ(Foo1B.quux);

new Foo1B();
new Foo1B.quux();

bar1B();
bar1B.quux();

XYZ(bar1B);
XYZ(bar1B.quux);

new bar1B();
new bar1B.quux();

//---

import {baz2A} from 'x2';
import {default as Foo2A, baz2B} from 'x2';
import Foo2B, {default as Foo2C, baz2C} from 'x2';

baz2A();
baz2A.quux();

XYZ(baz2A);
XYZ(baz2A.quux);

new baz2A();
new baz2A.quux();

Foo2A();
Foo2A.quux();

XYZ(Foo2A);
XYZ(Foo2A.quux);

new Foo2A();
new Foo2A.quux();

baz2B();
baz2B.quux();

XYZ(baz2B);
XYZ(baz2B.quux);

new baz2B();
new baz2B.quux();

Foo2B();
Foo2B.quux();

XYZ(Foo2B);
XYZ(Foo2B.quux);

new Foo2B();
new Foo2B.quux();

Foo2C();
Foo2C.quux();

XYZ(Foo2C);
XYZ(Foo2C.quux);

new Foo2C();
new Foo2C.quux();

baz2C();
baz2C.quux();

XYZ(baz2C);
XYZ(baz2C.quux);

new baz2C();
new baz2C.quux();

//---

import {baz3A as qux3A, default as Foo3A} from 'x3';

baz3A();
baz3A.quux();

XYZ(baz3A);
XYZ(baz3A.quux);

new baz3A();
new baz3A.quux();

qux3A();
qux3A.quux();

XYZ(qux3A);
XYZ(qux3A.quux);

new qux3A();
new qux3A.quux();

Foo3A();
Foo3A.quux();

XYZ(Foo3A);
XYZ(Foo3A.quux);

new Foo3A();
new Foo3A.quux();
