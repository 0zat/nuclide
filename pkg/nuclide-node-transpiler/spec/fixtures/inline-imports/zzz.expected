function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj['default'] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

require('x');

//---

var _x1;

function _load_x1() {
  return _x1 = _interopRequireDefault(require('x1'));
}

var _x12;

function _load_x12() {
  return _x12 = _interopRequireWildcard(require('x1'));
}

var _x13;

function _load_x13() {
  return _x13 = _interopRequireWildcard(require('x1'));
}

(0, (_x1 || _load_x1()).default)();
(_x1 || _load_x1()).default.quux();

XYZ((_x1 || _load_x1()).default);
XYZ((_x1 || _load_x1()).default.quux);

new (_x1 || _load_x1()).default();
new (_x1 || _load_x1()).default.quux();

(0, _x12 || _load_x12())();
(_x12 || _load_x12()).quux();

XYZ(_x12 || _load_x12());
XYZ((_x12 || _load_x12()).quux);

new (_x12 || _load_x12())();
new (_x12 || _load_x12()).quux();

(0, (_x13 || _load_x13()).default)();
(_x13 || _load_x13()).default.quux();

XYZ((_x13 || _load_x13()).default);
XYZ((_x13 || _load_x13()).default.quux);

new (_x13 || _load_x13()).default();
new (_x13 || _load_x13()).default.quux();

(0, _x13 || _load_x13())();
(_x13 || _load_x13()).quux();

XYZ(_x13 || _load_x13());
XYZ((_x13 || _load_x13()).quux);

new (_x13 || _load_x13())();
new (_x13 || _load_x13()).quux();

//---

var _x2;

function _load_x2() {
  return _x2 = require('x2');
}

var _x22;

function _load_x22() {
  return _x22 = _interopRequireDefault(require('x2'));
}

var _x23;

function _load_x23() {
  return _x23 = require('x2');
}

var _x24;

function _load_x24() {
  return _x24 = _interopRequireDefault(require('x2'));
}

var _x25;

function _load_x25() {
  return _x25 = require('x2');
}

(0, (_x2 || _load_x2()).baz2A)();
(_x2 || _load_x2()).baz2A.quux();

XYZ((_x2 || _load_x2()).baz2A);
XYZ((_x2 || _load_x2()).baz2A.quux);

new (_x2 || _load_x2()).baz2A();
new (_x2 || _load_x2()).baz2A.quux();

(0, (_x22 || _load_x22()).default)();
(_x22 || _load_x22()).default.quux();

XYZ((_x22 || _load_x22()).default);
XYZ((_x22 || _load_x22()).default.quux);

new (_x22 || _load_x22()).default();
new (_x22 || _load_x22()).default.quux();

(0, (_x23 || _load_x23()).baz2B)();
(_x23 || _load_x23()).baz2B.quux();

XYZ((_x23 || _load_x23()).baz2B);
XYZ((_x23 || _load_x23()).baz2B.quux);

new (_x23 || _load_x23()).baz2B();
new (_x23 || _load_x23()).baz2B.quux();

(0, (_x24 || _load_x24()).default)();
(_x24 || _load_x24()).default.quux();

XYZ((_x24 || _load_x24()).default);
XYZ((_x24 || _load_x24()).default.quux);

new (_x24 || _load_x24()).default();
new (_x24 || _load_x24()).default.quux();

(0, (_x24 || _load_x24()).default)();
(_x24 || _load_x24()).default.quux();

XYZ((_x24 || _load_x24()).default);
XYZ((_x24 || _load_x24()).default.quux);

new (_x24 || _load_x24()).default();
new (_x24 || _load_x24()).default.quux();

(0, (_x25 || _load_x25()).baz2C)();
(_x25 || _load_x25()).baz2C.quux();

XYZ((_x25 || _load_x25()).baz2C);
XYZ((_x25 || _load_x25()).baz2C.quux);

new (_x25 || _load_x25()).baz2C();
new (_x25 || _load_x25()).baz2C.quux();

//---

var _x3;

function _load_x3() {
  return _x3 = _interopRequireDefault(require('x3'));
}

var _x32;

function _load_x32() {
  return _x32 = require('x3');
}

baz3A();
baz3A.quux();

XYZ(baz3A);
XYZ(baz3A.quux);

new baz3A();
new baz3A.quux();

(0, (_x32 || _load_x32()).baz3A)();
(_x32 || _load_x32()).baz3A.quux();

XYZ((_x32 || _load_x32()).baz3A);
XYZ((_x32 || _load_x32()).baz3A.quux);

new (_x32 || _load_x32()).baz3A();
new (_x32 || _load_x32()).baz3A.quux();

(0, (_x3 || _load_x3()).default)();
(_x3 || _load_x3()).default.quux();

XYZ((_x3 || _load_x3()).default);
XYZ((_x3 || _load_x3()).default.quux);

new (_x3 || _load_x3()).default();
new (_x3 || _load_x3()).default.quux();
